minitest-rails.gemspec

```diff
@@ -15,10 +15,10 @@ Gem::Specification.new do |gem|
                       ["README.md", "UPDATING.md", "CHANGELOG.md", "LICENSE"]
   gem.require_paths = ["lib"]
 
-  gem.required_ruby_version = ">= 3.1.0"
+  gem.required_ruby_version = ">= 3.2.0"
 
   gem.add_dependency "minitest", "~> 5.20"
-  gem.add_dependency "railties", ">= 7.2.0", "< 8.0.0"
+  gem.add_dependency "railties", ">= 8.0.0", "< 8.1.0"
```

Comment:
Please don't, see: https://github.com/svenfuchs/rails-i18n/pull/1130#discussion_r1808623501

Reply:
I think minitest-rails is in a different situation than rails-i18n, since it aims to release with the same major and minor version as the Rails version it supports.

Reply:
Just an observation, because the commit doesn't say why. But the most recent activity like this chose to use `< [next major version].0.0` as the upper bound.
https://github.com/minitest/minitest-rails/commit/a3b07807652993910e681a633f2e350bfbbb17db#diff-b0f4e53f9bc79026fa5f7eb8b63bdf2c87232f0ba7b91918d5896531e65fa980R21

Reply:
@blowmage What would you prefer here? `< 8.1.0`, `< 9.0.0` or no upper limit at all?

Reply:
@pdobb the comment right below that one does explain why: "This upper dependency require to do the same dance every few years [list of PRs] That prevent people from trying Rails release candidates etc. It's just a bad pattern."

An upper limit causes needless delay and churn when things don't actually need a change. Let's assume we remove the upper constraint and Rails 9 comes out as an example:
- If nothing breaks with minitest-rails: great! No PR/gem release needed to unblock apps to upgrade to the newest version of Rails. This is an improvement over the current way.
- If something breaks with minitest-rails: we'll wait for a new release to fix it. This is the same as today.

In my app's Gemfile I can pin specific versions if I know that works better for my situation, but I can't override the upper constraint as easily.

Reply:
The intention of this PR is merely to add Rails 8.0 support to this gem, and I have tried to do it with as few changes as possible, to make it easy to review.

If you want minitest-rails to no longer have the upper version constraint on the railties version, you should feel free to open a separate pull request.